packages/devtools_app/test/flutter/network_profiler_test.dart

```diff
@@ -0,0 +1,302 @@
+        expect(find.byType(HttpRequestTimingView), findsOneWidget);
+        expect(find.byType(HttpRequestCookiesView), findsNothing);
+
+        // There should be a tile for each of the instant events, plus the
+        // total duration.
+        expect(
+          find.byType(ExpansionTile),
+          findsNWidgets(data.instantEvents.length + 1),
+        );
+      }
+
+      Future<void> validateCookiesTab(HttpRequestData data) async {
+        final hasCookies =
+            tableSource.currentSelectionListenable.value.hasCookies;
+
+        if (hasCookies) {
+          // Switch to cookies tab.
+          await tester.tap(find.byKey(HttpRequestInspector.cookiesTabKey));
+          await tester.pumpAndSettle();
+
+          expect(find.byType(HttpRequestHeadersView), findsNothing);
+          expect(find.byType(HttpRequestTimingView), findsNothing);
+          expect(find.byType(HttpRequestCookiesView), findsOneWidget);
+
+          // Checks the contents of a cookies table to ensure it's well formed.
+          void validateCookieTable(List<Cookie> cookies, Key key) {
+            expect(
+              find.byKey(key),
+              findsOneWidget,
+            );
+            final cookieCount = cookies.length;
+            final DataTable cookiesTable = tester.widget(
+              find.byKey(key),
+            );
+            expect(cookiesTable.rows.length, cookieCount);
+          }
+
+          // Check the request cookies table.
+          if (data.requestCookies.isNotEmpty) {
+            validateCookieTable(
+              data.requestCookies,
+              HttpRequestCookiesView.requestCookiesKey,
+            );
+          }
+
+          // Check the response cookies table.
+          if (data.responseCookies.isNotEmpty) {
+            validateCookieTable(
+              data.responseCookies,
+              HttpRequestCookiesView.responseCookiesKey,
+            );
+          }
+        } else {
+          // The cookies tab shouldn't be displayed if there are no cookies
+          // associated with the request.
+          expect(find.byKey(HttpRequestInspector.cookiesTabKey), findsNothing);
+        }
+      }
+
+      // Note: we only iterate over the first 25 rows as that's what's displayed
+      // by in the table.
+      for (final row in find
+          .byKey(HttpRequestDataTableSource.httpRequestRowKey)
+          .evaluate()) {
+        // Tap a row and ensure the inspector is populated.
+        await tester.tap(find.byWidget(row.widget));
+        await tester.pumpAndSettle();
+        expect(
+          find.byKey(HttpRequestInspector.noRequestSelectedKey),
+          findsNothing,
+        );
+
+        final selection = tableSource.currentSelectionListenable.value;
+        await validateHeadersTab(selection);
+        await validateTimingTab(selection);
+        await validateCookiesTab(selection);
+      }
+
+      // Clear the selection.
```

Comment:
why do we have to tap twice?

Reply:
This is to select a row and deselect. This was cleaner than keeping track of the last selected item and then deselecting it.